app/main.jsx

```diff
@@ -12,19 +12,25 @@ import ProductContainer from './containers/ProductContainer'
 // import WhoAmI from './components/WhoAmI'
 import App from './components/App'
 
-import {fetchProducts} from './reducers/products'
+import {fetchProducts, fetchProductById} from './reducers/products'
 
 function onProductsEnter() {
   store.dispatch(fetchProducts())
 }
 
+// do we really need to go fetch this info again? Maybe to get reviews....
```

Comment:
I'm definitely not 100% on this but I think this is why we always need to fetch the product information. If the product has changed from when the all products view was loaded to when they load the individual product view, then the current information for the needs to be displayed. Someone could load the all products view, and leave that view open for hours or even days. If there were any updates, then they need to be shown. I think it also breaks a principal of Redux, for a view to pass information to another view instead of getting the information from the state?